<!DOCTYPE html>
<html lang="en">
  <!-- Head -->
  <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-CVBCKTZY2X"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-CVBCKTZY2X');
</script>    <!-- Metadata, OpenGraph and Schema.org -->
    

    <!-- Standard metadata -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Analyze Genome Assembly | Inbar  Ofer</title>
    <meta name="author" content="Inbar  Ofer">
    <meta name="description" content="Analyze genome assembly from NGS reads&lt;br&gt;&lt;b&gt;Using: &lt;/b&gt;&lt;em&gt;bash, Trimmomatic, SPAdes, Quast&lt;/em&gt;">
    <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website">


    <!-- Bootstrap & MDB -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous">

    <!-- Fonts & Icons -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

    <!-- Code Syntax Highlighting -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light">

    <!-- Styles -->
    
    <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;">
    
    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="canonical" href="https://inbaro-neu.github.io/projects/analyze_assembly/">
    
    <!-- Dark Mode -->
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark">

    <script src="/assets/js/theme.js"></script>
    <script src="/assets/js/dark_mode.js"></script>
    

  </head>

  <!-- Body -->
  <body class="fixed-top-nav ">

    <!-- Header -->
    <header>

      <!-- Nav Bar -->
      <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
        <div class="container">
          <a class="navbar-brand title font-weight-lighter" href="/"><span class="font-weight-bold">InbarÂ </span>Ofer</a>
          <!-- Navbar Toggle -->
          <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar top-bar"></span>
            <span class="icon-bar middle-bar"></span>
            <span class="icon-bar bottom-bar"></span>
          </button>

          <div class="collapse navbar-collapse text-right" id="navbarNav">
            <ul class="navbar-nav ml-auto flex-nowrap">

              <!-- About -->
              <li class="nav-item ">
                <a class="nav-link" href="/">About</a>
              </li>
              

              <!-- Other pages -->
              <li class="nav-item ">
                <a class="nav-link" href="/projects/">Projects</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/resume/">Resume</a>
              </li>

              <!-- Toogle theme mode -->
              <li class="toggle-container">
                <button id="light-toggle" title="Change theme">
                  <i class="fas fa-moon"></i>
                  <i class="fas fa-sun"></i>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </nav>

      <!-- Scrolling Progress Bar -->
      <progress id="progress" value="0">
        <div class="progress-container">
          <span class="progress-bar"></span>
        </div>
      </progress>
    </header>


    <!-- Content -->
    <div class="container mt-5">
      <!-- page.html -->
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-CVBCKTZY2X"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-CVBCKTZY2X');
</script>
        <div class="post">

          <header class="post-header">
            <h1 class="post-title">Analyze Genome Assembly</h1>
            <p class="post-description">Analyze genome assembly from NGS reads<br><b>Using: </b><em>bash, Trimmomatic, SPAdes, Quast</em></p>
          </header>

          <article>
            <p><em>Author:</em> Inbar Ofer</p>

<ul>
  <li>Access to the GitHub repository may be provided upon request.</li>
</ul>

<h2 id="overview">Overview</h2>

<p>This program performs an analysis of a genome assembly from next-generation sequencing (NGS) reads. The sbatch_assembleGenome.conf is currently configured to assemble the <i>Rhodobacter spheroides</i> genome, with the SRA accession number of SRR522244.</p>

<h2 id="running-sbatch_assemblegenomesh">Running sbatch_assembleGenome.sh</h2>
<p>This program is run from the command line, but first you must load the Anaconda environment for BINF6308:</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load anaconda3/2021.11
source activate "BINF-12-2021"
</code></pre></div></div>
<p>Then, you can run the program by typing the following command:
<code class="language-plaintext highlighter-rouge">./sbatch_assembleGenome.sh</code>.</p>

<p>Alternatively, you can run the program by running each of the scripts individually:</p>
<ol>
  <li>
<code class="language-plaintext highlighter-rouge">source sbatch_assembleGenome.conf</code> in order to set the environment variables</li>
  <li>
<code class="language-plaintext highlighter-rouge">srun --partition=short --pty --export=ALL --nodes=1 --ntasks=1 --mem=10Gb --time=04:00:00 /bin/bash</code> in order to request a compute node on the Discovery remote server</li>
  <li><code class="language-plaintext highlighter-rouge">./getNGS.sh</code></li>
  <li><code class="language-plaintext highlighter-rouge">./trim.sh</code></li>
  <li><code class="language-plaintext highlighter-rouge">./runSpades.sh</code></li>
  <li><code class="language-plaintext highlighter-rouge">./runQuast.sh</code></li>
</ol>

<h2 id="methods">Methods</h2>
<p>The program uses the following scripts to assemble the genome:</p>

<h3 id="1-getngssh">1. getNGS.sh</h3>
<p>This program downloads the raw NGS data from the NCBI SRA database using fasterq-dump.</p>

<p>fasterq-dump is used to convert SRA files into the more commonly used FASTQ format, which is a widely accepted format for storing raw sequence data with associated quality scores. The command works faster than the original fastq-dump command provided in earlier versions of the SRA Toolkit.</p>

<p>getNGS.sh uses the following flags/options with fasterq-dump:</p>
<ul>
  <li>
<code class="language-plaintext highlighter-rouge">--split-3 ${SRR_ID}</code>: split the reads into two files, one for each end of the read</li>
  <li>
<code class="language-plaintext highlighter-rouge">-O "../data/untrimmed"</code>: output the files to the untrimmed directory</li>
</ul>

<h3 id="2-trimsh">2. trim.sh</h3>
<p>This program trims the raw NGS data using Trimmomatic, a widely used, flexible, and high-performance command-line Java-based tool designed to identify and remove low-quality bases, adapter sequences, and other technical artifacts that can negatively affect downstream analysis of sequencing data.</p>

<p>trim.sh uses the following flags/options with Trimmomatic:</p>
<ul>
  <li>
<code class="language-plaintext highlighter-rouge">PE</code> indicates that the input is paired-end (reads).</li>
  <li>
<code class="language-plaintext highlighter-rouge">-threads 1</code> indicates that the program should use 1 thread.</li>
  <li>
<code class="language-plaintext highlighter-rouge">-phred33</code> indicates that the quality scores are encoded using ASCII+33.</li>
  <li>The next two parameters are the left and right read files, respectively.</li>
  <li>The next four parameters are the output files for paired and unpaired output. The unpaired output files contain reads where the mate was deleted due to overall low quality. Since you always have to have the same number of reads in the left file and right file for paired-end reads, if one read is deleted entirely due to low quality, its mate has to be deleted as well. The reads whose mates have been deleted end up in the unpaired files.</li>
  <li>
<code class="language-plaintext highlighter-rouge">HEADCROP:0</code> indicates that the none of the first bases of each read should be removed.</li>
  <li>
<code class="language-plaintext highlighter-rouge">ILLUMINACLIP:$PATH_TO_TRIMMOMATIC/adapters/TruSeq3-PE.fa:2:30:10</code> indicates that the TruSeq3-PE.fa file should be used to remove adapters. The first number indicates the maximum mismatch count which will still allow a full match to be performed. The second number indicates the minimum score of the match between any adapter and the read. The third number indicates that if a full match is not found, the algorithm will check for a partial match with the next lowest score. If the score is higher than the second number, the match will be performed. The second and third numbers are there to prevent the algorithm from removing parts of the read that happen to match the adapter.</li>
  <li>
<code class="language-plaintext highlighter-rouge">LEADING:20</code> indicates that any base with a quality score less than 20 should be removed from the beginning of the read.</li>
  <li>
<code class="language-plaintext highlighter-rouge">TRAILING:20</code> indicates that any base with a quality score less than 20 should be removed from the end of the read.</li>
  <li>
<code class="language-plaintext highlighter-rouge">SLIDINGWINDOW:4:30</code> indicates that any window of 4 consecutive bases where the average quality score is less than 30 should be removed.</li>
  <li>
<code class="language-plaintext highlighter-rouge">MINLEN:36</code> indicates that any read that is less than 36 bases long should be removed.</li>
</ul>

<h3 id="3-runspadessh">3. runSpades.sh</h3>
<p>This program assembles the genome using SPAdes (St. Petersburg genome assembler) and the trimmed, paired reads extracted using getNGS.sh and trim.sh. SPAdes is a widely used, open-source genome assembly tool designed for de novo assembly of small genomes, such as bacteria, fungi, and viruses, using high-throughput sequencing data.</p>

<p>runSpades.sh uses the following flags/options with SPAdes:</p>
<ul>
  <li>
<code class="language-plaintext highlighter-rouge">-1</code> indicates the first read file that should be used.</li>
  <li>
<code class="language-plaintext highlighter-rouge">-2</code> indicates the second read file that should be used.</li>
  <li>
<code class="language-plaintext highlighter-rouge">-o</code> indicates the output directory for the assembly results.</li>
</ul>

<h3 id="4-runquastsh">4. runQuast.sh</h3>
<p>This program runs Quast to evaluate the assembly. QUAST (Quality Assessment Tool for Genome Assemblies) is an open-source, user-friendly tool designed to evaluate and provide statistics for genome assemblies. QUAST generates a comprehensive report that includes various assembly metrics, such as N50 and L50 statistics, total length, and number of contigs.</p>

<p>runQuast.sh uses the following flags/options with Quast:</p>
<ul>
  <li>The first parameter is the reference genome.</li>
  <li>
<code class="language-plaintext highlighter-rouge">-o</code> indicates the output directory.</li>
</ul>

<h2 id="conclusions-from-analysis">Conclusions from Analysis</h2>

<p><b>Download Quast report: </b><a href="/assets/pdf/quast_report.pdf" target="_blank" rel="noopener noreferrer"><i class="fas fa-file-pdf" style="font-size: 1.5rem"></i></a></p>

<p><b>What were the key metrics?</b></p>

<p>The key metrics from the Quast report were:</p>
<ul>
  <li>N50: 25,496</li>
  <li>L50: 51</li>
  <li>Number of contigs: 323</li>
  <li>Total length: 4,531,150</li>
</ul>

<p><b>How âgoodâ was the assembly?</b></p>

<p>The N50 value is a popular way to guage an assemblyâs quality. In simple terms, itâs the length of the shortest contig in a set where the fewest (biggest) contigs make up at least half of the assembly. When the N50 value is higher, itâs usually a good sign because it means the contigs are longer and the assembly is more contiguous.</p>

<p>The L50 value counts the contigs that, when added together, make up at least half of the assembly. If the L50 value is lower, itâs generally better because it means fewer contigs are needed to cover half of the assembly.</p>

<p>With an N50 value of 25,496 and an L50 value of 51, this assembly seems pretty contiguous, with a decent amount of longer contigs. However, there is still room for improvement, as an even higher N50 value and a lower L50 value would mean an even more contiguous assembly.</p>

<p>Another way to gauge the assemblyâs quality is by looking at the number of contigs and the total length of the assembly. Ideally, you want a smaller number of contigs and a total length close to the anticipated genome size.</p>

<p>The assemblyâs contig count (323) is relatively low, which is a positive sign because it means the genome is represented by fewer, longer segments.</p>

<p><i>Rhodobacter sphaeroides</i>, the bacterium in question, has a genome size of around 4.6 million base pairs. So, since the total length of the assembly is close to that, itâs a good indicator that the assembly likely covers most, if not all, of the genome.</p>

<p>Overall, this assembly seems to be of reasonable quality, with a total length near the expected genome size and a fairly low number of contigs. However, thereâs still a bit of room for improvement when it comes to contiguity, as the N50 and L50 values suggest.</p>

          </article>

        </div>

    </div>

    <!-- Footer -->    
    <footer class="fixed-bottom">
      <div class="container mt-0">
        Â© Copyright 2023 Inbar  Ofer. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Photos from <a href="https://unsplash.com" target="_blank" rel="external nofollow noopener">Unsplash</a>.

      </div>
    </footer>

    <!-- JavaScripts -->
    <!-- jQuery -->
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

    <!-- Bootsrap & MDB scripts -->
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script>

    <!-- Masonry & imagesLoaded -->
  <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
  <script defer src="/assets/js/masonry.js" type="text/javascript"></script>
    
  <!-- Medium Zoom JS -->
  <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script>
  <script defer src="/assets/js/zoom.js"></script><!-- Load Common JS -->
  <script defer src="/assets/js/common.js"></script>
  <script defer src="/assets/js/copy_code.js" type="text/javascript"></script>

    
  <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
  <script async src="https://badge.dimensions.ai/badge.js"></script>

    <!-- MathJax -->
  <script type="text/javascript">
    window.MathJax = {
      tex: {
        tags: 'ams'
      }
    };
  </script>
  <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script>
  <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>

    
    

<!-- Scrolling Progress Bar -->
<script type="text/javascript">
  /*
   * This JavaScript code has been adapted from the article 
   * https://css-tricks.com/reading-position-indicator/ authored by Pankaj Parashar, 
   * published on the website https://css-tricks.com on the 7th of May, 2014.
   * Couple of changes were made to the original code to make it compatible 
   * with the `al-foio` theme.
   */
  const progressBar = $("#progress");
  /*
   * We set up the bar after all elements are done loading.
   * In some cases, if the images in the page are larger than the intended
   * size they'll have on the page, they'll be resized via CSS to accomodate
   * the desired size. This mistake, however, breaks the computations as the
   * scroll size is computed as soon as the elements finish loading.
   * To account for this, a minimal delay was introduced before computing the
   * values.
   */
  window.onload = function () {
    setTimeout(progressBarSetup, 50);
  };
  /*
   * We set up the bar according to the browser.
   * If the browser supports the progress element we use that.
   * Otherwise, we resize the bar thru CSS styling
   */
  function progressBarSetup() {
    if ("max" in document.createElement("progress")) {
      initializeProgressElement();
      $(document).on("scroll", function() {
        progressBar.attr({ value: getCurrentScrollPosition() });
      });
      $(window).on("resize", initializeProgressElement);
    } else {
      resizeProgressBar();
      $(document).on("scroll", resizeProgressBar);
      $(window).on("resize", resizeProgressBar);
    }
  }
  /*
   * The vertical scroll position is the same as the number of pixels that
   * are hidden from view above the scrollable area. Thus, a value > 0 is
   * how much the user has scrolled from the top
   */
  function getCurrentScrollPosition() {
    return $(window).scrollTop();
  }

  function initializeProgressElement() {
    let navbarHeight = $("#navbar").outerHeight(true);
    $("body").css({ "padding-top": navbarHeight });
    $("progress-container").css({ "padding-top": navbarHeight });
    progressBar.css({ top: navbarHeight });
    progressBar.attr({
      max: getDistanceToScroll(),
      value: getCurrentScrollPosition(),
    });
  }
  /*
   * The offset between the html document height and the browser viewport
   * height will be greater than zero if vertical scroll is possible.
   * This is the distance the user can scroll
   */
  function getDistanceToScroll() {
    return $(document).height() - $(window).height();
  }

  function resizeProgressBar() {
    progressBar.css({ width: getWidthPercentage() + "%" });
  }
  // The scroll ratio equals the percentage to resize the bar
  function getWidthPercentage() {
    return (getCurrentScrollPosition() / getDistanceToScroll()) * 100;
  }
</script>

  </body>
</html>
