<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>Transcriptome Assembly | Inbar Ofer</title> <meta name="author" content="Inbar Ofer"> <meta name="description" content="Automate guided and de novo transcriptome assembly&lt;br&gt;&lt;b&gt;Using: &lt;/b&gt;&lt;em&gt;bash, Trinity, Slurm&lt;/em&gt;"> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700%7CRoboto+Slab:100,300,400,500,700%7CMaterial+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="https://inbaro-neu.github.io/projects/transcriptome_assembly/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js"></script> <script src="/assets/js/dark_mode.js"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"><span class="font-weight-bold">Inbar </span>Ofer</a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">About</a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">Projects</a> </li> <li class="nav-item "> <a class="nav-link" href="/resume/">Resume</a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fas fa-moon"></i> <i class="fas fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title">Transcriptome Assembly</h1> <p class="post-description">Automate guided and de novo transcriptome assembly<br><b>Using: </b><em>bash, Trinity, Slurm</em></p> </header> <article> <p><em>Author:</em> Inbar Ofer</p> <ul> <li>Access to the GitHub repository may be provided upon request.</li> </ul> <h2 id="overview">Overview</h2> <p>This program is a bash script designed to automate the guided and de novo transcriptome assembly using Trinity for RNA sequencing data. The script includes Slurm directives for resource allocation, software module loading, working directory setup, and file copying for the analysis.</p> <ol> <li>Specify Slurm parameters.</li> <li>Load the necessary software modules and create a log directory.</li> <li>Copy the trimmed fastq data and alignment files to the working directory.</li> <li>Run guided assembly using Trinity by merging all BAM alignment files and assembling the guided transcriptome.</li> <li>Analyze the guided transcriptome and output the stats.</li> <li>Run de novo assembly using Trinity and assemble the de novo transcriptome.</li> <li>Analyze the de novo transcriptome and output the stats.</li> <li>Move the output files to the user’s home directory.</li> </ol> <h2 id="running-sbatch_trinitysh">Running sbatch_trinity.sh</h2> <p><code class="language-plaintext highlighter-rouge">sbatch sbatch_trinity.sh</code></p> <h2 id="methods">Methods</h2> <h3 id="mergeallsh">mergeAll.sh</h3> <p>This bash script is designed to merge multiple sorted BAM files using the <code class="language-plaintext highlighter-rouge">samtools</code> [1] program. The merged BAM file can be used for downstream analysis, such as transcriptome assembly.</p> <p><code class="language-plaintext highlighter-rouge">samtools</code> is used with the following parameters:</p> <ul> <li> <code class="language-plaintext highlighter-rouge">merge</code>: the subcommand to merge multiple BAM files into a single file.</li> <li> <code class="language-plaintext highlighter-rouge">-b data/bam/bamIn.txt</code>: the <code class="language-plaintext highlighter-rouge">-b</code> option specifies that the input files are listed in a text file, <code class="language-plaintext highlighter-rouge">bamIn.txt</code>, which contains the paths to the input files. This option is used to avoid command-line length limits when merging a large number of files.</li> <li> <code class="language-plaintext highlighter-rouge">data/bam/bamIn.txt</code>: the path to the input file that contains the list of sorted BAM files to be merged.</li> <li> <code class="language-plaintext highlighter-rouge">data/bam/AipAll.bam</code>: the output file name of the merged BAM file. The output file is stored in the data/bam directory.</li> </ul> <h3 id="runtrinitysh">runTrinity.sh</h3> <p>This bash script is designed to perform a guided assembly using Trinity [2], a de novo transcriptome assembler, with the aid of the genome-aligned RNA-Seq reads in a BAM format.</p> <p>Trinity is used with the following parameters:</p> <ul> <li> <code class="language-plaintext highlighter-rouge">--genome_guided_bam data/bam/AipAll.bam</code>: the path to the BAM file containing the genome-aligned RNA-Seq reads.</li> <li> <code class="language-plaintext highlighter-rouge">--genome_guided_max_intron 10000</code>: the maximum intron length allowed in the genome-guided assembly. In this case, the maximum intron length is set to 10000 bp.</li> <li> <code class="language-plaintext highlighter-rouge">--max_memory 10G</code>: the maximum amount of memory that Trinity can use for the assembly process. In this case, the maximum memory is set to 10 GB.</li> <li> <code class="language-plaintext highlighter-rouge">--CPU 4</code>: the number of CPUs/threads to be used for the assembly process. In this case, 4 CPUs are used.</li> <li> <code class="language-plaintext highlighter-rouge">--output results/trinity_guided</code>: the output directory where the results of the guided assembly will be stored.</li> </ul> <h3 id="analyzetrinitysh">analyzeTrinity.sh</h3> <p>This bash script is designed to generate statistics for the guided transcriptome assembly output by Trinity.</p> <p>The script runs TrinityStats.pl, a Perl script that generates assembly statistics for Trinity output. The script takes the following input:</p> <ul> <li> <code class="language-plaintext highlighter-rouge">results/trinity_guided/Trinity-GG.fasta</code>: the path to the guided transcriptome assembly output by Trinity.</li> </ul> <h3 id="trinitydenovosh">trinityDeNovo.sh</h3> <p>This bash script is designed to perform a de novo transcriptome assembly using Trinity [2] with the paired-end RNA-Seq reads in FASTQ format.</p> <p>Trinity is used with the following parameters:</p> <ul> <li> <code class="language-plaintext highlighter-rouge">--seqType fq</code>: the type of input sequence data, which in this case is paired-end FASTQ.</li> <li> <code class="language-plaintext highlighter-rouge">--output results/trinity_de_novo</code>: the output directory where the results of the de novo assembly will be stored.</li> <li> <code class="language-plaintext highlighter-rouge">--max_memory 10G</code>: the maximum amount of memory that Trinity can use for the assembly process. In this case, the maximum memory is set to 10 GB.</li> <li> <code class="language-plaintext highlighter-rouge">--CPU 4</code>: the number of CPUs/threads to be used for the assembly process. In this case, 4 CPUs are used.</li> <li> <code class="language-plaintext highlighter-rouge">--left $leftReads</code>: the path to the left reads in FASTQ format. The variable $leftReads contains a list of paths to the left reads that are obtained using the ls command.</li> <li> <code class="language-plaintext highlighter-rouge">--right $rightReads</code>: the path to the right reads in FASTQ format. The variable $rightReads contains a list of paths to the right reads that are obtained using the ls command.</li> </ul> <h3 id="analyzetrinitydenovosh">analyzeTrinityDeNovo.sh</h3> <p>This bash script is designed to generate statistics for the de novo transcriptome assembly output by Trinity.</p> <p>The script runs TrinityStats.pl, a Perl script that generates assembly statistics for Trinity output. The script takes the following input:</p> <ul> <li> <code class="language-plaintext highlighter-rouge">results/trinity_de_novo/Trinity.fasta</code>: the path to the de novo transcriptome assembly output by Trinity.</li> </ul> <h2 id="assembly-statistics">Assembly Statistics</h2> <p>The de novo assembly has more Trinity genes and transcripts than the guided assembly, which could indicate that it was able to capture more of the transcriptome. However, the guided assembly may have a more accurate representation of the transcriptome because it was guided by an existing reference genome. The higher N50 value in the guided assembly could indicate a more contiguous assembly, which is generally considered to be an indicator of higher quality.</p> <h2 id="references">References</h2> <ol> <li> <p>Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M. O., Whitwham, A., Keane, T., McCarthy, S. A., Davies, R. M., &amp; Li, H. (2021). Twelve years of SAMtools and BCFtools. <em>Gigascience</em>, <em>10</em>(2). <a href="https://doi.org/10.1093/gigascience/giab008" rel="external nofollow noopener" target="_blank">https://doi.org/10.1093/gigascience/giab008</a>.</p> </li> <li> <p>Grabherr, M. G., Haas, B. J., Yassour, M., Levin, J. Z., Thompson, D. A., Amit, I., … &amp; Regev, A. (2011). Full-length transcriptome assembly from RNA-Seq data without a reference genome. <em>Nature biotechnology</em>, <em>29</em>(7), 644-652. <a href="https://doi.org/10.1038/nbt.1883" rel="external nofollow noopener" target="_blank">https://doi.org/10.1038/nbt.1883</a>.</p> </li> </ol> </article> </div> </div> <footer class="fixed-bottom"> <div class="container mt-0"> © Copyright 2023 Inbar Ofer. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Photos from <a href="https://unsplash.com" target="_blank" rel="external nofollow noopener">Unsplash</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js"></script> <script defer src="/assets/js/common.js"></script> <script defer src="/assets/js/copy_code.js" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>